src/manifolds/Sphere.jl

```diff
@@ -84,59 +149,61 @@ function check_tangent_vector(
     return nothing
 end
 
-decorated_manifold(M::Sphere) = Euclidean(representation_size(M)...; field=ℝ)
+decorated_manifold(M::AbstractSphere{𝔽}) where {𝔽} = Euclidean(representation_size(M)...; field=𝔽)
+get_embedding(M::AbstractSphere{𝔽}) where {𝔽} = decorated_manifold(M)
+
 
 @doc raw"""
-    distance(M::Sphere, p, q)
+    distance(M::AbstractSphere, p, q)
 
-Compute the geodesic distance betweeen `p` and `q` on the [`Sphere`](@ref) `M`.
+Compute the geodesic distance betweeen `p` and `q` on the [`AbstractSphere`](@ref) `M`.
 The formula is given by the (shorter) great arc length on the (or a) great circle
 both `p` and `q` lie on.
 
 ````math
-d_{𝕊^n}(p,q) = \arccos(⟨p,q⟩).
+d_{𝕊}(p,q) = \arccos(⟨p,q⟩).
 ````
 """
-distance(::Sphere, x, y) = acos(clamp(dot(x, y), -1, 1))
+distance(::AbstractSphere, p, q) = acos(clamp(real(dot(p, q)), -1, 1))
 
-embed!(::Sphere, q, p) = (q .= p)
+embed!(::AbstractSphere, q, p) = (q .= p)
 
-embed!(::Sphere, Y, p, X) = (Y .= X)
+embed!(::AbstractSphere, Y, p, X) = (Y .= X)
 
 @doc raw"""
-    exp(M::Sphere, p, X)
+    exp(M::AbstractSphere, p, X)
 
-Compute the exponential map from `p` in the tangent direction `X` on the [`Sphere`](@ref)
+Compute the exponential map from `p` in the tangent direction `X` on the [`AbstractSphere`](@ref)
 `M` by following the great arc eminating from `p` in direction `X`.
 
 ````math
 \exp_p X = \cos(\lVert X \rVert_p)p + \sin(\lVert X \rVert_p)\frac{X}{\lVert X \rVert_p}X,
 ````
-where $\lVert X \rVert_p$ is the [`norm`](@ref norm(::Sphere,p,X)) on the
-[`Sphere`](@ref) `M`.
+where $\lVert X \rVert_p$ is the [`norm`](@ref norm(::AbstractSphere,p,X)) on the
+tangent space at `p` of the [`AbstractSphere`](@ref) `M`.
 """
-exp(::Sphere, ::Any...)
+exp(::AbstractSphere, ::Any...)
 
-function exp!(M::Sphere, q, p, X)
+function exp!(M::AbstractSphere, q, p, X)
     θ = norm(M, p, X)
     q .= cos(θ) .* p .+ usinc(θ) .* X
     return q
 end
 
-flat!(M::Sphere, ξ::CoTFVector, p, X::TFVector) = copyto!(ξ, X)
+flat!(M::AbstractSphere, ξ::CoTFVector, p, X::TFVector) = copyto!(ξ, X)
 
-function get_basis(M::Sphere{N}, p, B::DiagonalizingOrthonormalBasis) where {N}
-    A = zeros(N + 1, N + 1)
+function get_basis(M::Sphere{n,ℝ}, p, B::DiagonalizingOrthonormalBasis{T,ℝ}) where {T,n}
+    A = zeros(n+1, n+1)
     A[1, :] = transpose(p)
     A[2, :] = transpose(B.frame_direction)
     V = nullspace(A)
-    κ = ones(N)
+    κ = ones(n)
     if !iszero(B.frame_direction)
         # if we have a nonzero direction for the geodesic, add it and it gets curvature zero from the tensor
         V = cat(B.frame_direction / norm(M, p, B.frame_direction), V; dims = 2)
         κ[1] = 0 # no curvature along the geodesic direction, if x!=y
     end
-    Ξ = [V[:, i] for i = 1:N]
+    Ξ = [V[:, i] for i = 1:manifold_dimension(M)]
```

Comment:
```suggestion
    Ξ = [V[:, i] for i = 1:n]
```
This can still be just `n` like before.